### PR TITLE
Remove the "Func" name, use "Asynchronous" and "Observable"

### PR DESCRIPTION
**Pull request by reyang**
Follow up https://github.com/open-telemetry/opentelemetry-specification/pull/1617#issuecomment-821454608.

This has been discussed and concluded during the [metrics API/SDK spec SIG meeting](https://docs.google.com/document/d/1-bCYkN-DWJq4jw1ybaDZYYmx-WAe6HnwfWbkm8d57v8/edit#).

## Changes

* Used "Asynchronous Counter" as the name of the entry.
* Added description to encourage implementations to use "ObservableCounter".
* Clarified that these concepts have nothing to do with asynchronous pattern or observer pattern.
* Changed other async instruments so we have consistency.

### CODE REVIEW

**Comment by reyang** [comment]
@jonatan-ivanov FYI.

File: specification/metrics/new_api.md [hunk 1]
```diff
@@ -405,7 +412,7 @@ def pf_callback():
         (10465,    ("pid", 880), ("bitness", 32)),
     )
 
-page_faults_counter_func = meter.create_counter_func(name="PF", description="process page faults", pf_callback)
+page_faults_observable_counter = meter.create_observable_counter(name="PF", description="process page faults", pf_callback)
```
Comment (jkwatson) [inline specification/metrics/new_api.md:415]: what is being returned here? is there a need to have a handle to something in this python API?

Comment (reyang) [reply specification/metrics/new_api.md:415]: We don't have a clear spec on the return value yet, I think it is depending on the Hint API.
If we only allow hints to be provided during the instrument creation, then we probably don't need a return value (unless certain languages require a handle for dispose/dtor).
If we allow hints to be added post the instrument definition, we might need to return a handle.

Comment (reyang) [reply specification/metrics/new_api.md:415]: For example (this is something we'll have to figure out later):

Hints required during instrument creation:
```python
# no return value
meter.create_observable_counter(
    name="PF",
    description="process page faults",
    hint={"default_dimensions": ("pid",)},
    pf_callback,
)
```

Hints required after instrument creation:
```python
# no return value
page_faults_observable_counter = meter.create_observable_counter(
    name="PF",
    description="process page faults",
    pf_callback,
)
page_faults_observable_counter.AddHint(default_dimensions=("pid",))
```

Comment (jkwatson) [reply specification/metrics/new_api.md:415]: Until we have something to use the return value for, I'd recommend leaving it out of these examples.

Comment (reyang) [reply specification/metrics/new_api.md]: Good point, I've updated the PR.

**Review by victlu — APPROVED** [review]
LGTM

**Review by tarekgh — APPROVED** [review]

File: specification/metrics/new_api.md [hunk 2]
```diff
@@ -512,32 +519,32 @@ TODO
...
 
 TODO
 
-#### UpDownCounterFunc operations
+#### Asynchronous UpDownCounter operations
 
-`UpDownCounterFunc` is only intended for asynchronous scenario. The only operation is
-provided by the `callback`, which is registered during the [UpDownCounterFunc
-creation](#updowncounterfunc-creation).
+Asynchronous UpDownCounter is only intended for asynchronous scenario. The only operation is
```
Comment (jkwatson) [inline specification/metrics/new_api.md:545]:
```suggestion
Asynchronous UpDownCounter is only intended for an asynchronous scenario. The only operation is
```

File: specification/metrics/new_api.md [hunk 3]
```diff
@@ -462,31 +469,31 @@ TODO
...
 
 TODO
 
-#### GaugeFunc operations
+#### Asynchronous Gauge operations
 
-`GaugeFunc` is only intended for asynchronous scenario. The only operation is
-provided by the `callback`, which is registered during the [GaugeFunc
-creation](#gaugefunc-creation).
+Asynchronous Gauge is only intended for asynchronous scenario. The only operation is
```
Comment (jkwatson) [inline specification/metrics/new_api.md:494]:
```suggestion
Asynchronous Gauge is only intended for an asynchronous scenario. The only operation is
```

File: specification/metrics/new_api.md [hunk 4]
```diff
@@ -432,14 +439,14 @@ interface IAtomicClock
...
+meter.CreateObservableCounter<UInt64>("caesium_oscillates", () => clock.GetCaesiumOscillates());
 ```
 
-#### CounterFunc operations
+#### Asynchronous Counter operations
 
-`CounterFunc` is only intended for asynchronous scenario. The only operation is
-provided by the `callback`, which is registered during the [CounterFunc
-creation](#counterfunc-creation).
+Asynchronous Counter is only intended for asynchronous scenario. The only
```
Comment (jkwatson) [inline specification/metrics/new_api.md:447]:
```suggestion
Asynchronous Counter is only intended for an asynchronous scenario. The only
```

**Review by jkwatson — APPROVED** [review]
Thanks!

**Review by jsuereth — APPROVED** [review]

**Review by jonatan-ivanov — APPROVED** [review]

**Review by jmacd — APPROVED** [review]

**Review by alolita — COMMENTED** [review]
lgtm.

**Review by weyert — APPROVED** [review]

**Review by noahfalk — APPROVED** [review]
Thanks!

**Review by bogdandrutu — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Pull request merged**